src/ORM/DataObject.php

```diff
@@ -940,6 +949,50 @@ public function i18n_plural_name()
         return _t(static::class . '.PLURALNAME', $this->plural_name());
     }
 
+    /**
+     * Get description for this class
+     * @return null|string
+     */
+    public function classDescription()
+    {
+        return static::config()->get('class_description', Config::UNINHERITED);
+    }
+
+    /**
+     * Get localised description for this class
+     * @return null|string
+     */
+    public function i18n_classDescription()
+    {
+        $placeholder = 'PLACEHOLDER_DESCRIPTION';
+        $baseDescription = $this->classDescription() ?? $placeholder;
+
+        // Check the new i18n key first
+        $description = _t(static::class . '.CLASS_DESCRIPTION', $baseDescription);
+        if ($description !== $baseDescription) {
+            return $description;
+        }
+
+        // Fall back on the deprecated localisation key
+        $legacyI18n = _t(static::class . '.DESCRIPTION', $baseDescription);
+        if ($legacyI18n !== $baseDescription) {
+            Deprecation::notice(
+                '5.4.0',
+                'The ' . static::class . '.DESCRIPTION' . ' i18n localisation key is deprecated. Use '
+                    . static::class . '.CLASS_DESCRIPTION' . ' instead.',
+                Deprecation::SCOPE_GLOBAL
+            );
```

Comment:
```suggestion
            $descKey = static::class . '.DESCRIPTION';
            $classDescKey = static::class . '.CLASS_DESCRIPTION';
            Deprecation::notice(
                '5.4.0',
                "The $deskKey i18n localisation key is deprecated. Use $classDescKey instead."
            );
```
Should use SCOPE_CLASS (the default) not SCOPE_GLOBAL?

Reply:
The class isn't deprecated, so `SCOPE_CLASS` isn't appropriate. `SCOPE_CLASS` would mean the deprecation notice says something like "DataObject is deprecated" which just isn't true.

Reply:
Also `SCOPE_METHOD` is the default - and is also not appropriate because the method isn't deprecated either.

Reply:
Right. I don't think we should be deprecating translation keys even though this is a high use area. I don't think we've ever done this before as we'll simply stop using old translation keys. It seems like its setting a new precedent/policy which I don't think needs to be there.

The new keys are just duplicates of the old keys, so the only risk of something going wrong is that any new translation updates won't be received.

If you really want to keep this, then you'll need to update the CMS 6 PR to throw exceptions if the old keys are used otherwise there seems like there is very little point in doing this

Reply:
We don't throw exceptions when things are deprecated. We use deprecation notices. We clearly need to discuss this in person lol it seems like this is being blown out of proportion a bit.

Reply:
Agree offline to keep the fallback though remove the deprecation notices

Reply:
Done